tests/Unit/Helpers/Evolution/DiscontinuousGalerkin/BoundaryConditions.hpp

```diff
@@ -0,0 +1,624 @@
+                "All boundary condition classes must be marked `final`.");
+  using variables_tags = typename System::variables_tag::tags_list;
+  using flux_variables = typename System::flux_variables;
+  using fluxes_tags =
+      db::wrap_tags_in<::Tags::Flux, flux_variables, tmpl::size_t<FaceDim + 1>,
+                       Frame::Inertial>;
+  const std::unique_ptr<domain::BoundaryConditions::BoundaryCondition>
+      boundary_condition =
+          TestHelpers::test_factory_creation<BoundaryConditionBase>(
+              factory_string);
+
+  REQUIRE_FALSE(
+      domain::BoundaryConditions::is_periodic(boundary_condition->get_clone()));
+
+  tmpl::for_each<BoundaryCorrectionsList>(
+      [&boundary_condition, &box_of_volume_data, epsilon, &face_points,
+       &generator, &python_boundary_condition_functions, &python_module,
+       &ranges](auto boundary_correction_v) {
+        using BoundaryCorrection =
+            tmpl::type_from<decltype(boundary_correction_v)>;
+        using package_data_input_tags = tmpl::append<
+            variables_tags, fluxes_tags,
+            typename BoundaryCorrection::dg_package_data_temporary_tags,
+            typename ::evolution::dg::Actions::detail::get_primitive_vars<
+                System::has_primitive_and_conservative_vars>::
+                template f<BoundaryCorrection>>;
+        using boundary_condition_dg_gridless_tags =
+            typename BoundaryCondition::dg_gridless_tags;
+        for (const auto use_moving_mesh : {false, true}) {
+          detail::test_boundary_condition_with_python_impl<
+              System, ConversionClassList, BoundaryCorrection>(
+              generator, python_module, python_boundary_condition_functions,
+              dynamic_cast<const BoundaryCondition&>(*boundary_condition),
+              face_points, box_of_volume_data, ranges, use_moving_mesh,
+              variables_tags{}, package_data_input_tags{},
+              boundary_condition_dg_gridless_tags{}, epsilon);
+          // Now serialize and deserialize and test again
+          INFO("Test boundary condition after serialization.");
+          const auto deserialized_bc =
+              serialize_and_deserialize(boundary_condition);
+          detail::test_boundary_condition_with_python_impl<
+              System, ConversionClassList, BoundaryCorrection>(
+              generator, python_module, python_boundary_condition_functions,
+              dynamic_cast<const BoundaryCondition&>(*deserialized_bc),
+              face_points, box_of_volume_data, ranges, use_moving_mesh,
+              variables_tags{}, package_data_input_tags{},
+              boundary_condition_dg_gridless_tags{}, epsilon);
+        }
+      });
+}
+
+/// Test that a boundary condition is correctly identified as periodic.
+template <typename BoundaryCondition, typename BoundaryConditionBase>
+void test_periodic_condition(const std::string& factory_string) {
```

Comment:
For my understanding: should/will there be a similar helper for None BCs?

Reply:
Yes, I think there should be, just to make life easier :) Maybe I'll wait until the GH system to add that since that's a system where we will actually need the `None` condition. Does that sound okay?

Reply:
Yes 👍